can/message.py

```diff
@@ -290,29 +303,29 @@ def equals(
         :param timestamp_delta: the maximum difference at which two timestamps are
                                 still considered equal or None to not compare timestamps
 
-        :return: True iff the given message equals this one
+        :param check_channel: compare the channels of the messages
+
+        :return: True if the given message equals this one
         """
         # see https://github.com/hardbyte/python-can/pull/413 for a discussion
         # on why a delta of 1.0e-6 was chosen
+        if self is other:
+            return True
+        if type(self) != type(other):  # pylint: disable=unidiomatic-typecheck
+            return False
```

Comment:
It’s probably very unlikely but what if someone subclasses can.Message? Shouldn’t that also be possible to compare with?

Reply:
NO.
Do you have experience on other OO high level languagey like Java, Go, Haskell?

Reply:
Allright, I guess it make sense.